charms/worker/k8s/src/charm.py

```diff
@@ -694,8 +722,9 @@ def _announce_kubernetes_version(self):
                 if not unit_version:
                     raise ReconcilerError(f"Waiting for version from {unit.name}")
                 if unit_version != local_version:
-                    status.add(ops.BlockedStatus(f"Version mismatch with {unit.name}"))
-                    raise ReconcilerError(f"Version mismatch with {unit.name}")
+                    # NOTE: Add a check to validate if we are doing an upgrade
+                    status.add(ops.WaitingStatus("Upgrading the cluster"))
+                    return
```

Comment:
Ahh, i see now that it's because the raising of the reconciler error prevented the upgrade events from engaging.   Whew -- we still need a check to let folks know they're running on out-of-spec-version of the applications

For now `_announce_kubernetes_version` is only run on the lead CP.   say you deployed and related a kw 1.35 to a 1.31 cluster.  I imagine the 1.35 workers may not join.  Should they join?  Is the k8s-cp the right place to gripe about it?  You're right that we should at least make sure we're not in an upgrade scenario before we raise the reconciler error.